testing/test_optimizers.py

```diff
@@ -175,6 +176,25 @@ def test_optimizer_interrupt(self):
                                              "non-succesfull result expected.")
         self.assertTrue(np.allclose(result.x, 0.0), msg="The optimum will not be identified nonetheless")
 
+    def test_failsafe(self):
+        X, Y = self.optimizer.acquisition.data[0], self.optimizer.acquisition.data[1]
+        # Provoke cholesky faillure
+        self.optimizer.acquisition._optimize_restarts = 1
+        self.optimizer.acquisition.models[0].likelihood.variance.transform = GPflow.transforms.Identity()
+        self.optimizer.acquisition.models[0].likelihood.variance = -5.0
+        self.optimizer.acquisition.models[0]._needs_recompile = True
+        with self.assertRaises(RuntimeError) as e:
+            with self.optimizer.failsafe():
+                self.optimizer.acquisition.set_data(X, Y)
+
+        fname = 'failed_bopt_{0}.npz'.format(id(e.exception))
+        self.assertTrue(os.path.isfile(fname))
+        data = np.load(fname)
+        np.testing.assert_almost_equal(data['X'], X)
+        np.testing.assert_almost_equal(data['Y'], Y)
+        print(os.remove(fname))
```

Comment:
remove does not return any value? can't find it in the python docs.

if it does, this probably is a leftover and we can make an assert of it?